packages/http-client-python/emitter/src/emitter.ts

```diff
@@ -37,10 +37,7 @@ function addDefaultOptions(sdkContext: PythonSdkContext) {
     // if they pass in a flavor other than azure, we want to ignore the value
     (options as any).flavor = undefined;
   }
-  if (
-    (options as any).flavor === undefined &&
-    sdkContext.emitContext.emitterOutputDir.includes("azure")
-  ) {
+  if (sdkContext.sdkPackage.crossLanguagePackageId.toLowerCase().includes("azure")) {
```

Comment:
`crossLanguagePackageId` is full namespace in typespec: https://github.com/Azure/typespec-azure/blob/fea410950e6bef0eddd0a4387aa874e20dec1637/packages/typespec-client-generator-core/src/public-utils.ts#L298. It is too general compared with old logic but acceptable to me.